lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -856,6 +856,109 @@ SwiftLanguage::GetHardcodedSynthetics() {
+      auto type = valobj.GetCompilerType();
+      auto tsc = type.GetTypeSystem().dyn_cast_or_null<TypeSystemClang>();
+      if (!tsc)
+        return nullptr;
+
+      ProcessSP process_sp(valobj.GetProcessSP());
+      auto *swift_runtime = SwiftLanguageRuntime::Get(process_sp);
+      if (!swift_runtime)
+        return nullptr;
+
+      auto module = valobj.GetModule();
+      if (!module) {
+        module = valobj.GetTargetSP()
+                     ->GetExecutableModule();
+        if (!module)
+          return nullptr;
+      }
+
+      auto gts = module->GetTypeSystemForLanguage(
+          lldb::LanguageType::eLanguageTypeSwift);
+      if (!gts)
+        return nullptr;
+
+      auto *ts = llvm::cast<TypeSystemSwift>(gts->get());
+
+      // We operate directly on the qualifier type because the TypeSystem
+      // interface doesn't allow us to check for static constexpr members.
+      auto qual_type = TypeSystemClang::GetQualType(type.GetOpaqueQualType());
+      auto *record_type = llvm::dyn_cast_or_null<clang::RecordType>(
+          qual_type.getTypePtrOrNull());
+      if (!record_type)
+        return nullptr;
+
+      const clang::RecordDecl *record_decl = record_type->getDecl();
+      CompilerType swift_type;
+
+      for (auto *child_decl : record_decl->decls()) {
+        auto *var_decl = llvm::dyn_cast<clang::VarDecl>(child_decl);
+        if (!var_decl)
+          continue;
+
+        auto name = var_decl->getName();
+        if (name != "$__swift_mangled_name")
+          continue;
+
+        const auto *typedef_type =
+            llvm::dyn_cast<clang::TypedefType>(var_decl->getType());
+        if (!typedef_type)
+          continue;
+
+        auto *decl = typedef_type->getDecl();
+        if (!decl)
+          continue;
+
+        auto swift_name = decl->getName();
+        if (!swift::Demangle::isMangledName(swift_name))
+          continue;
+
+        swift_type = ts->GetTypeFromMangledTypename(ConstString(swift_name));
+        break;
+      }
+
+      if (!swift_type)
+        return nullptr;
+
+      auto swift_valobj =
+          SwiftLanguageRuntime::ExtractSwiftValueObjectFromCxxWrapper(valobj);
+      if (!swift_valobj)
+        return nullptr;
+      // Cast it to a Swift type since thhe swift runtime expects a Swift value object.
+      auto casted = swift_valobj->Cast(swift_type);
+      if (casted) {
+        TypeAndOrName type_or_name;
+        Address address;
+        Value::ValueType value_type;
+        // Try to find the dynamic type of the Swift type.
+        // TODO: find a way to get the dyamic value type from the 
+        // command.
+        if (swift_runtime->GetDynamicTypeAndAddress(
+                *casted.get(), lldb::DynamicValueType::eDynamicCanRunTarget,
+                type_or_name, address, value_type)) {
+          swift_type = type_or_name.GetCompilerType();
+          // Cast it to the more specific type.
+          casted = casted->Cast(swift_type);
```

Comment:
GetDynamicTypeAndAddress can return true even though it only found a NAME for the dynamic type, not the type.  For instance, you might be debugging a library that takes in BaseClass * being used by code for which you don't have debug info that passes in a DerivedFromBaseClass *.  It's still useful to know that that is the dynamic type, but in that case your TypeAndOrName will only have a Name, your GetType will be an invalid compiler type.  You should check that here.